osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/SmartMeterService.java

```diff
@@ -84,17 +86,27 @@ public SmartMeter getSmartMeter(final String deviceId, final SmartMeteringDevice
     }
 
     private ProtocolInfo getProtocolInfo(final SmartMeteringDevice smartMeteringDevice) throws FunctionalException {
-        final ProtocolInfo protocolInfo = this.protocolInfoRepository.findByProtocolAndProtocolVersion(
-                smartMeteringDevice.getProtocolName(), smartMeteringDevice.getProtocolVersion());
+
+        String protocolName = getProtocolName(smartMeteringDevice);
+
+        final ProtocolInfo protocolInfo = this.protocolInfoRepository
+                .findByProtocolAndProtocolVersion(protocolName, smartMeteringDevice.getProtocolVersion());
         if (protocolInfo == null) {
             throw new FunctionalException(FunctionalExceptionType.UNKNOWN_PROTOCOL_NAME_OR_VERSION,
                     ComponentType.DOMAIN_SMART_METERING);
         }
         return protocolInfo;
     }
 
-    private DeviceModel
-            getDeviceModel(final org.opensmartgridplatform.domain.core.valueobjects.DeviceModel deviceModel) {
+    private String getProtocolName(final SmartMeteringDevice smartMeteringDevice) {
+        if (CDMA.equals(smartMeteringDevice.getCommunicationMethod())) {
```

Comment:
Shouldn't this logic be defined in SmartMeteringDevice?

Reply:
good point, although this is a plain "value object/DTO" I can add a more DDD approach here.

Reply:
in this case not. The protocol name is SMR, DSMR etc, but in this case it needs to be appended with CDMA to lookup the protocol info. 
If the logic was moved to the domain object/DTO. then the protocol name would be overwritten with a non existing value. 
To make it more clear, I will change the method name to getProtocolLookupName.

Reply:
yes, that would be an improvement. Done.